FHE/Ciphertext.cpp

```diff
@@ -4,17 +4,39 @@
 
 #include "Math/modp.hpp"
 
-Ciphertext::Ciphertext(const FHE_PK& pk) : Ciphertext(pk.get_params())
+/**
+ * FFI Exports
+ */
+
+unique_ptr<Ciphertext> add_ciphertexts(const Ciphertext &c0, const Ciphertext &c1)
 {
+  unique_ptr<Ciphertext> result(new Ciphertext(c0.get_params()));
+  add(*result, c0, c1);
+
+  return result;
 }
 
+unique_ptr<Ciphertext> mul_ciphertexts(const Ciphertext &c0, const Ciphertext &c1, const FHE_PK &pk)
+{
+  unique_ptr<Ciphertext> result(new Ciphertext(c0.get_params()));
+  mul(*result, c0, c1, pk);
+
+  return result;
+}
 
-void Ciphertext::set(const Rq_Element& a0, const Rq_Element& a1,
-        const FHE_PK& pk)
+/**
+ * Implementation
+ */
+
+Ciphertext::Ciphertext(const FHE_PK &pk) : Ciphertext(pk.get_params())

```

Comment:
Do we actually use this constructor anywhere? Looks like e.g. above, we just use the `Ciphertext(const FHE_Params &p)` constructor

Reply:
yes this is used almost everywhere in the codebase